src/future/ReactiveValue.js

```diff
@@ -23,6 +23,10 @@ const createReactiveValue = defaultValue => {
   }
 
   function getCurrentValueAt(key) {
+    if (key === undefined) {
```

Comment:
So we can get the value for the whole form in `AutoFields` without passing down props.

Reply:
I also just realized, I think there's a more ominous problem here. I think there's a mismatch between the value you're getting and what's being subscribed to. At the very least, this makes the code hard to follow. But worse, there will likely be memory leaks or improper notifications. I think we should keep value based subscriptions separate from key based subscriptions so it's easy to follow what's happening and avoid these problems. We'll probably also want to add some tests at some point to guarantee subscriptions are cleaned up properly. Assuming we can do that without sneaking into implementation details. 🤔

Reply:
Ahh good points @jdeal. I was thinking just about getting the values of the form, but I agree, this also opens a can of worms.